RFC-0038-cpython-support.md

```diff
@@ -0,0 +1,102 @@
+- Temporarily for enablement work (for example while a new CPython version is being enabled).
+- CD for docker and binaries should run for all supported versions.
+- Wheel/release smoke test should run on all supported versions.
+
+
+### Detailed CPython new version enablement
+
+- Enable new version basic compilation
+  - When: Once the first beta version is released (Usually in May)
+  - Planned for 3.13: 5/07
+  - ETA: 1-2 weeks. Before beta 2
+  - Goal: Fix compilation issues allowing to compile PyTorch locally. Report any needed patch in CPython before beta 2.
+  - Who: Core Maintainers for pytorch/pytorch + relevant maintainers for managed domains
+  - Note: If a patch is needed in CPython, it will be made available easily until next beta release happens
+- Enable new version CI/CD Infrastructure
+  - When: As soon as basic compilation is done
+  - ETA: 2-3 weeks. 
+  - Goal: Generate nightly wheels and scaffolding for new version testing
+  - Who: Core Infra
+- High risk enablement work for the new version  for submodules
+  - When: as soon as the CI/CD infrastructure is done
+  - ETA: before CPython first RC
+  - Planned for 3.13: 07/30
+  - Goal: Verify high risk systems and report any needed patch in CPython such that they can be fixed before RC1
+  - Who: High risk systems owners. As of today this is Dynamo C++ code and python binding subsystems.
+- Low risk enablement work for the new version  for submodules
+  - When: as soon as the CI/CD infrastructure is done
+  - ETA: before CPython final RC
+  - Planned for 3.13: 09/03
+  - Goal: Enable all testing for the new CPython version
+  - Who: Core Maintainers handle the long tail of issues. Specific submodule owner handle larger migration efforts (dynamo, TorchScript, etc)
+- Full new version wheel release
+  - When: When the new cpython verion is officially released (Usually in October)
+  - Planned for 3.13: 10/01
+  - ETA: 2 week
+  - Goal: Update nightlies to track the final release and advertise it as fully working
+  - Who: Infra + Comms
+- Full new version conda release
+  - When: When the new cpython version and our runtime dependencies are supported onconda
+  - ETA: 2 week
+  - Goal: Push binaries that are officially supported
+  - Who: Infra + Comms
+
+
+### Detailed CPython old version Sunset
+
+- Deprecate old version
+  - When: During the release process of the last PyTorch release before the oldest version of python goes EOL
+  - Planned for 3.8: October
+  - Goal: announce deprecation of the oldest cpython version on dev-discuss + release
+  - Who: Comms
+- Stop releasing nightly binaries for oldest version
+  - When: Once the last supported release happened
+  - ETA: 2 weeks
+  - Goal: remove all wheel and conda binaries for the EOL version
+  - Who: Infra
+- Upgrade oldest version CI
+  - When: After nightlies are not published anymore
+  - ETA: 2 weeks
+  - Goal: migrate all the CI jobs running on the oldest version to the soon-to-be oldest version
+  - Who: Infra
+
```

Comment:
Would be nice to add a paragraph here about BE work associated with removing workarounds specific to older version (i.e. if any `if sys.version < (x, y)` in Python code and similar ifdefs in C++)